ccpa_cmplibrary/src/main/java/com/sourcepoint/ccpa_cmplibrary/CCPAConsentLib.java

```diff
@@ -147,7 +147,7 @@ private ConsentWebView buildWebView() {
             public void onMessageReady() {
                 Log.d("msgReady", "called");
                 if (mCountDownTimer != null) mCountDownTimer.cancel();
-                if(!onMessageReadyCalled) {
+                if(!onMessageReadyCalled && !hasErrorOccurred()) {
```

Comment:
What if instead of checking for a state (`hasErrorOccurred`) we don't set the callbacks to a "no-op"? This way we can safely call the callbacks without thrown exception *and* we don't need to keep track of state.

Reply:
The problem is that we don't wanna call onConsentUIReady if an error has occurred prior to it. Or do we? It would also work fine to just take this !hasErrorOccurred() out, but we cannot get rid of onMessageReadyCalled state so easily...
Besides, the problem was setting the callback to null [here](https://github.com/SourcePointUSA/CCPA_Android_SDK/blob/e11f689f37dc8151b49f01aa2bc8cb5f5d64450a/ccpa_cmplibrary/src/main/java/com/sourcepoint/ccpa_cmplibrary/CCPAConsentLib.java#L328) to simply getting rid of this line should already  solve it!

Reply:
done